lib/backend/k8s/conversion/conversion.go

```diff
@@ -212,6 +213,29 @@ func (c Converter) PodToWorkloadEndpoint(pod *kapiv1.Pod) (*model.KVPair, error)
 	labels[apiv3.LabelNamespace] = pod.Namespace
 	labels[apiv3.LabelOrchestrator] = apiv3.OrchestratorKubernetes
 
+	// Pull out floating IP annotation
+	var floatingIPs []apiv3.IPNAT
+	if annotation, ok := pod.Annotations["cni.projectcalico.org/floatingIPs"]; ok {
+
+		// Parse Annotation data
+		var ips []string
+		err := json.Unmarshal([]byte(annotation), &ips)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse '%s' as JSON: %s", annotation, err)
+		}
+
+		podip, _, err := cnet.ParseCIDROrIP(ipNets[0])
```

Comment:
I think Felix further validates that this is an IP, and not a CIDR so you should validate that it is an IP, not a CIDR here (or validate that it is a full-length /32 or /128).

Reply:
Wouldn't that always be the case with ParseCIDRorIP? It'll return net.ip / net.ipnet / err.

Reply:
No, ParseCIDROrIP will accept `10.1.2.0/24` and return `10.1.2.0` even though there's a `/24` on there.  The `/24` will fail validation later on because Felix is expecting a bare IP address.

Also, dropping the `/24` and passing through `10.1.2.0` doesn't seem right to me.